File: apps/server/internal/generic/roles.go
// Package generic holds shared types with no business logic of their own:
// RBAC role constants and per-domain sentinel errors. Nothing here depends
// on any other internal package, so every domain can import it freely
// without risking an import cycle.
package generic

// Role is one of the roles an organization member can hold. Exactly one
// per membership — these are better-auth's organization-plugin custom
// role names (see apps/web/lib/auth-access-control.ts), which the JWT
// plugin bakes into the "role" claim at issue time (see
// apps/web/lib/auth.ts's definePayload).
type Role string

const (
	RoleSuperAdmin Role = "super_admin"
	RoleAdmin      Role = "admin"
	RoleScanner    Role = "scanner"
)

File: apps/web/lib/roles.ts
// Single source of truth for organization role names — mirrors
// apps/server/internal/generic/roles.go. Keep these two files in lockstep:
// Go's RequireRole() compares JWT claims against exactly these strings.
export const ROLE_SUPER_ADMIN = "super_admin"
export const ROLE_ADMIN = "admin"
export const ROLE_SCANNER = "scanner"

export type OrgRole = typeof ROLE_SUPER_ADMIN | typeof ROLE_ADMIN | typeof ROLE_SCANNER
